src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -226,10 +226,27 @@ def create_connect_args(self, url):
             opts["host"] = opts["host"] + ".snowflakecomputing.com"
             opts["port"] = "443"
         opts["autocommit"] = False  # autocommit is disabled by default
-        opts.update(url.query)
+
+        cache_column_metadata = opts.pop("cache_column_metadata", None)
```

Comment:
What about setting using `"False"` as the default value and set ` self._cache_column_metadata =  parse_url_boolean(opts.pop("cache_column_metadata", "False"))`?

Reply:
It's a minor difference, I just prefer the code to not do parsing on constants if possible.